### PR TITLE
[Balance] fetch historical balance

### PR DESCRIPTION
**Pull request by xinyu-li-cb**
- init
- update tests with usd_value
- tests
- api.ts updates
- fix test
- update change log
- [Balance] fetch historical balance
- formatting
- fix lint
- add max size for fetching all history
- rebase error
- tweak naming
- address comments
- move list historical balance test under address
- address comment
- fix lint
- fix IDE change
- update changelog
- [PSDK-361] Gasless Sends Support (#142)
- add struct for ListHistoricalBalancesOptions

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>1</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>2</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,14 @@
 # Coinbase Node.js SDK Changelog
 
-## [0.0.15] 
+## [0.0.16]
 
-### Added 
+### Added
+
+- Add Function listHistoricalBalances for Adress for fetching historical balances for an asset
```
Comment (John-peterson-coinbase) [inline CHANGELOG.md:7]:
```suggestion
- Add Function `listHistoricalBalances` for `Address` for fetching historical balances for an asset
```

File: src/coinbase/historical_balance.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+   */
+  private constructor(amount: Decimal, blockHeight: Decimal, blockHash: string, asset: Asset) {
+    this.amount = amount;
+    this.blockHeight = blockHeight;
+    this.blockHash = blockHash;
+    this.asset = asset;
+  }
+
+  /**
+   * Converts a BalanceModel into a Balance object.
```
Comment (John-peterson-coinbase) [inline src/coinbase/historical_balance.ts:30]:
```suggestion
   * Converts a HistoricalBalanceModel into a HistoricalBalance object.
```

File: src/tests/historical_balance_test.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        asset_id: Coinbase.assets.Eth,
+        network_id: Coinbase.networks.BaseSepolia,
+        decimals: 18,
+        contract_address: "0x",
+      },
+    };
+
+    const historicalBalance = HistoricalBalance.fromModel(historyModel);
+
+    it("returns a new Balance object with the correct amount", () => {
```
Comment (John-peterson-coinbase) [inline src/tests/historical_balance_test.ts:23]:
```suggestion
    it("returns a new HistoricalBalance object with the correct amount", () => {
```

File: src/tests/historical_balance_test.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      expect(historicalBalance.amount).toEqual(amount);
+    });
+
+    it("returns a new Balance object with the correct asset_id", () => {
```
Comment (John-peterson-coinbase) [inline src/tests/historical_balance_test.ts:27]:
```suggestion
    it("returns a new HistoricalBalance object with the correct asset_id", () => {
```

**Review by John-peterson-coinbase — APPROVED** [review]

**Pull request merged**